moveos/moveos-stdlib/src/natives/moveos_stdlib/object.rs

```diff
@@ -113,7 +110,20 @@ fn borrow_object_reference(
         "Expected a struct type with layout same as Object<T>"
     );
 
-    //TODO should we keep the GlobalValue in Context?
+    let object_id = ObjectID::from_runtime_value(object_id_value)
+        .map_err(|_e| partial_extension_error("Invalid object id argument"))?;
+    let table_context = context.extensions_mut().get_mut::<NativeTableContext>();
+
+    let data = table_context.table_data();
+    let mut table_data = data.write();
+    let gv = table_data.get_or_create_object_reference(object_id.clone())?;
+
+    if gv.reference_count() >= 2 {
```

Comment:
Need to differentiate between `ref` and `mut ref`? Multiple `refs` should be reasonable

Reply:
The GlobalValue does not distinguish between ref and mut ref. I record an issue.